### PR TITLE
Add latest version check

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add periodic check for new version of Snowflake CLI on PyPi. The check is performed once every hour.

### CODE REVIEW

File: src/snowflake/cli/_app/version_check.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from snowflake.cli.__about__ import VERSION
+from snowflake.cli.api.console import cli_console
+
+
+def _get_latest_cli_version() -> Version | None:
+    try:
+        headers = {"Content-Type": "application/vnd.pypi.simple.v1+json"}
+        response = requests.get(
+            "https://pypi.org/pypi/snowflake-cli-labs/json", headers=headers
+        )
```
Comment (sfc-gh-mraba) [inline src/snowflake/cli/_app/version_check.py:12]: A timeout would be nice

File: src/snowflake/cli/_app/version_check.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        }
+        self._cache_file.write_text(json.dumps(data))
+
+    @staticmethod
+    def _get_version_from_pypi() -> str | None:
+        headers = {"Content-Type": "application/vnd.pypi.simple.v1+json"}
+        response = requests.get(
+            "https://pypi.org/pypi/snowflake-cli-labs/json", headers=headers, timeout=3
+        )
+        response.raise_for_status()
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/_app/version_check.py:51]: I think we shouldn't exit when any error happened, we could log it.

File: tests/app/test_version_check.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+from io import BytesIO
+from unittest.mock import patch
+
+from packaging.version import Version
+from requests import Response
+from snowflake.cli._app.version_check import _VersionCache, get_new_version_msg
+
```
Comment (sfc-gh-astus) [inline tests/app/test_version_check.py:7]: Missing tests:
- Error during get request not exit from app
- Message will not appear when cli_console is off

Comment (sfc-gh-turbaszek) [reply tests/app/test_version_check.py:7]: Added both tests

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**